modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/search/request/enhancer/body/contributor/GeneralSXPSearchRequestBodyContributorTest.java

```diff
@@ -44,6 +44,35 @@ public void setUp() throws Exception {
 		_searchRequestBuilder = searchRequestBuilderFactory.builder();
 	}
 
+	@Test
+	public void testIgnoreSearchableAssetTypesIfEmpty() {
+		List<String> entryClassNames = new ArrayList<>();
+
+		entryClassNames.add("com.liferay.journal.model.JournalArticle");
+
+		_searchRequestBuilder.withSearchContext(
+			searchContext -> searchContext.setEntryClassNames(
+				entryClassNames.toArray(new String[1])));
+
+		GeneralConfiguration generalConfiguration = new GeneralConfiguration();
+
+		generalConfiguration.setSearchableAssetTypes(new String[0]);
+
+		Configuration configuration = new Configuration();
+
+		configuration.setGeneralConfiguration(generalConfiguration);
+
+		_generalSXPSearchRequestBodyContributor.contribute(
+			configuration, _searchRequestBuilder, null);
+
+		SearchRequest searchRequest = _searchRequestBuilder.build();
+
+		Assert.assertEquals(
+			entryClassNames, searchRequest.getEntryClassNames());
+		Assert.assertEquals(
+			new ArrayList<>(), searchRequest.getModelIndexerClassNames());
```

Comment:
Check if the searchRequest.getModelIndexerClassNames() returns an empty list instead of comparing it to a new ArrayList